dbt/include/hive/macros/adapters.sql

```diff
@@ -258,4 +257,20 @@
   {% endcall %}
   {% do return(load_result('list_views_without_caching').table) %}
 {% endmacro %}
-k
+
+{% macro get_hive_version() %}
+  {% set version_results = run_query('select version()') %}
+  {% if execute %}
+     {% set num_rows = (version_results | length) %}
+
+     {% if num_rows == 1 %}
+        {% set version_text = version_results[0][0] %}
```

Comment:
can u give an sample of how version_results/version_text would look in case on hive 2 and hive 3

Reply:
Sample for Hive 2:
<pre>
0: jdbc:hive2://localhost:10000> select version();
+--------------------------------------------------+
|                       _c0                        |
+--------------------------------------------------+
| 2.3.2 r857a9fd8ad725a53bd95c1b2d6612f9b1155f44d  |
+--------------------------------------------------+
</pre>

Sample for Hive 3:
<pre>
+--------------------------------------------------+
|                       _c0                        |
+--------------------------------------------------+
| 3.1.3000.7.2.15.1-1 r386e2de0b8d5cb3f1803ec755243d43185505de6 |
+--------------------------------------------------+
</pre>